common/src/main/java/org/opensearch/sql/common/utils/StringUtils.java

```diff
@@ -27,16 +27,44 @@ public static String unquote(String text, String mark) {
 
   /**
    * Unquote Identifier which has " or ' or ` as mark.
+   * Replace doubled quotes within the string with single version
    * @param text string
    * @return An unquoted string whose outer pair of (single/double/back-tick) quotes have been
    *     removed
    */
   public static String unquoteText(String text) {
-    if (isQuoted(text, "\"") || isQuoted(text, "'") || isQuoted(text, "`")) {
-      return text.substring(1, text.length() - 1);
-    } else {
+    StringBuilder textSB = new StringBuilder();
+
+    char enclosingQuote = Character.MIN_VALUE;
+    char firstChar = text.charAt(0);
+    char lastChar = text.charAt(text.length() - 1);
+
+    if (firstChar == lastChar
+        && (firstChar == '\''
+        || firstChar == '"'
+        || firstChar == '`')) {
+      enclosingQuote = firstChar;
+    }
+
+    if (enclosingQuote == Character.MIN_VALUE) {
       return text;
     }
+    char currentChar;
+    char nextChar;
+
+    // Ignores first and last character as they are the quotes that should be removed
+    for (int chIndex = 1; chIndex < text.length() - 1; chIndex++) {
+      currentChar = text.charAt(chIndex);
+      nextChar = text.charAt(chIndex + 1);
+      if (currentChar == enclosingQuote
```

Comment:
What happens if I pass: `'hello''`?  Should this pass?  Or `'don't'`?

Reply:
Errors in sql, passes through StringUtils as expected by removing quotes around string but keeping internal single quote.